### PR TITLE
Chat: Enable nostr search by npub 

### PR DESCRIPTION
**Pull request by abhay-raizada**
Objective: Make nostr profile searchable and clickable

TODO (as seperate PR's):

1. Improve chat interface design.
2. Make chat functionally work

UI:

image

image

### CODE REVIEW

**Comment by islandbitcoin** [comment]
tried to run it but got an error on pressing start button.

error

**Comment by islandbitcoin** [comment]
maybe a missing dependency?

**Comment by abhay-raizada** [comment]
@islandbitcoin n i didn't handle the case where nostrSecretKey would not be an nsec unless it's fetched from the localstorage or generated, so i cannot attempt to derive an npub from it, my bad, fixed it. didn't show up for me probably cause the key was already loaded in memory.

File: app/components/chat-message/chat-message.tsx [hunk 1]
```diff
@@ -136,25 +133,29 @@ export const ChatMessage: React.FC<Props> = ({ sender, seckey, recipient, messag
...
+    ) {
       const senderSec: secureKeyPair = {
-        seckey: seckey?.key,
-        pubkey: sender?.hexpubkey(),
+        seckey: nip19.decode(senderSecretKey).data.toString(),
+        pubkey: nip19.decode(senderPubKey).data.toString(),
       }
       const recipientSec: secureKeyPair = {
         seckey: "39c34c3f2600a36d582cf9fca1bfffc102f0532d4c1ba74a2d1aa5afcb061c31",
         pubkey: "7ebc48d3e51a4d81719b7c1d601fd6f55b08e83d731e2eda27aacc543cbc1840",
```
Comment (islandbitcoin) [inline app/components/chat-message/chat-message.tsx:147]: Do we still need this?

File: app/hooks/use-nostr-profile.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
-import { nip19, generateSecretKey } from "nostr-tools"
+import { nip19, generateSecretKey, getPublicKey, SimplePool } from "nostr-tools"
 
 const useNostrProfile = () => {
   const KEYCHAIN_NOSTRCREDS_KEY = "nostr_creds_key"
   const [nostrSecretKey, setNostrSecretKey] = useState<string>("")
+  const relays = [
+    "wss://relay.damus.io",
+    "wss://relay.primal.net",
+    "wss://nostr.pleb.network",
```
Comment (islandbitcoin) [inline app/hooks/use-nostr-profile.ts:11]: you can remove this one "nostr.pleb.network" - its offline and we will be changing the domain name

**Review by islandbitcoin — CHANGES_REQUESTED** [review]
Small changes

**Comment by islandbitcoin** [comment]
Performance note:
when I search for an npub, it takes a few seconds to load before showing me the result. is it possible to load the list in the background so the search is faster?

**Comment by Nodirbek75** [comment]
Getting this error when send message: 
Screenshot_1709045180

**Comment by Nodirbek75** [comment]
This is how it is displaying on android:
Screenshot_1709045562

I think "No messages here yet" is showing wrong.

**Review by Nodirbek75 — CHANGES_REQUESTED** [review]
Need to fix above mentioned issue and change the logo.
If possible, make chat modal better

File: app/components/chat-message/chat-message.tsx (hunk 1 above)
Comment (abhay-raizada) [reply app/components/chat-message/chat-message.tsx:147]: no, but i've kept it will update it in my chat window re-work PR

**Comment by abhay-raizada** [comment]
@Nodirbek75 I've opened a new issue for chat window changes https://github.com/lnflash/flash-mobile/issues/100 ,

right now the scope of this change is only till: https://github.com/lnflash/flash-mobile/issues/63

**Review by Nodirbek75 — APPROVED** [review]

**Review by islandbitcoin — APPROVED** [review]
ACK

**Pull request merged**